plasmapy/utils/tests/test_checks.py

```diff
@@ -5,108 +5,124 @@
 import pytest
 
 from ...constants import c
-from ..checks import (_check_quantity, _check_relativistic)
+from ..checks import (check_quantity, check_relativistic)
 
 
 def test__check_quantity():
+    def run_test(arg_value, units, can_be_negative=True, can_be_complex=False, can_be_inf=True):
+        """ Tests the check_quantity decorator
+        Tests that the given value is an astropy Quantity with correct units
+        and valid numerical values
+        Builds a method, decorates it, and immediately calls it
+        """
+        @check_quantity({
+            'arg': {'units': units, 'can_be_negative': can_be_negative, 'can_be_complex': can_be_complex, 'can_be_inf': can_be_inf}
+        })
+        def test_method(arg): pass
```

Comment:
Why is `run_test` defining a wrapped function and then calling it? Could you not achieve the same thing by decorating `run_test` instead?